### PR TITLE
enhancement: Combine all requirements into single file

### PR DESCRIPTION
**Pull request by CagriYonca**
Combined all requirements into a single file, and added python version conditions.

### CODE REVIEW

File: tests/requirements.txt [hunk 1]
```diff
@@ -4,28 +4,36 @@ boto3>=1.17.74
...
+fastapi>=0.92.0; python_version < "3.13"
 flask>=2.3.2
-grpcio>=1.37.1
-google-cloud-pubsub>=2.0.0
-google-cloud-storage>=1.24.0
+# grpc is not supported on 3.13 yet:
+# https://github.com/grpc/grpc/issues/34922
+grpcio>=1.37.1; python_version < "3.13"
+# depends grpcio on 3.13
+google-cloud-pubsub>=2.0.0; python_version < "3.12"
```
Comment (pvital) [inline tests/requirements.txt:18]: Shouldn't it be 3.13?
```suggestion
google-cloud-pubsub>=2.0.0; python_version < "3.13"
```

File: tests/requirements.txt [hunk 2]
```diff
@@ -4,28 +4,36 @@ boto3>=1.17.74
...
+# depends grpcio on 3.13
+google-cloud-storage>=1.24.0; python_version < "3.12"
```
Comment (pvital) [inline tests/requirements.txt:20]:
```suggestion
google-cloud-storage>=1.24.0; python_version < "3.13"
```

File: tests/requirements.txt [hunk 3]
```diff
@@ -4,28 +4,36 @@ boto3>=1.17.74
...
+legacy-cgi>=2.6.1; python_version == "3.13"
 lxml>=4.9.2
+markupsafe>=2.1.0;python_version >= "3.10"
```
Comment (pvital) [inline tests/requirements.txt:23]: What are the versions for Python 3.8 and 3.9?

File: tests/requirements.txt [hunk 4]
```diff
@@ -34,8 +42,11 @@ pytz>=2024.1
 redis>=3.5.3
 requests-mock
 responses<=0.17.0
-sanic>=19.9.0
-sanic-testing>=24.6.0
+# Sanic is not installable on 3.13 because `httptools, uvloop` dependencies fail to compile:
+# `too few arguments to function ‘_PyLong_AsByteArray’`
+sanic>=19.9.0; python_version < "3.13"
+sanic-testing>=24.6.0; python_version < "3.13"
+starlette>=0.38.2; python_version == "3.13"
```
Comment (pvital) [inline tests/requirements.txt:49]:
```suggestion
starlette>=0.12.13
```

**Review by pvital — CHANGES_REQUESTED** [review]
Few requests and comments.

**Review by GSVarsha — APPROVED** [review]
LGTM

**Review by pvital — APPROVED** [review]
It looks good to me.

**Pull request merged**